pkg/controller/job/controller.go

```diff
@@ -113,6 +119,24 @@ func NewJobController(podInformer framework.SharedInformer, kubeClient clientset
 		framework.ResourceEventHandlerFuncs{
 			AddFunc: jm.enqueueController,
 			UpdateFunc: func(old, cur interface{}) {
+				oldJob := old.(*batch.Job)
+				curJob := cur.(*batch.Job)
+
+				// We should invalidate the whole lookup cache if a Job's selector has been updated.
```

Comment:
Actually that should not happen, since selector is one of the immutable fields for a job.